sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/V2TableReference.scala

```diff
@@ -120,7 +121,10 @@ private[sql] object V2TableReferenceUtils extends SQLConfHelper {
       ctx: TemporaryViewContext): Unit = {
     val tableName = ref.identifier.toQualifiedNameParts(ref.catalog)
 
-    val dataErrors = V2TableUtil.validateCapturedColumns(table, ref.info.columns)
+    val dataErrors = V2TableUtil.validateCapturedColumns(
+      table,
+      ref.info.columns,
+      mode = ALLOW_NEW_TOP_LEVEL_FIELDS)
```

Comment:
This should be the SQL view behavior.

Reply:
ideally we should allow both new columns and fields. But the implementation is non-trivial as we need to rewrite the struct instead of a simple project list.